### PR TITLE
Update social-logos to include new reddit icon

### LINKED ISSUE #9819: Sharing: update Reddit button to match brand guidelines
The Reddit share button is outdated. It doesn't fit Reddit's current brand guidelines. The button should be orange and the logo needs to be updated. Right now the button is blue and it's using an older logo. 

Brand guidelines: https://www.reddithelp.com/sites/default/files/2018-03/reddit_brand_guidelines.pdf

Logo download: https://www.reddithelp.com/sites/default/files/2018-03/reddit_logos.zip

### LINKED ISSUE #12439: Sharing: Make Reddit button orange to be in line with their brand guidelines
The Reddit button orange to be in line with their brand guidelines.

Screenshot from 2019-05-22 20 28 29

The light blue one is also rather not attractive.

### PR DESCRIPTION
**Pull request by brbrr**
I've used official Reddit asset and modified it a bit: removed background circle and cleared out all styling within svg file. These changes might need to be confirmed wih them. As per their brand book:

> Reddit permits a few specific uses without prior approval. For those uses, if you agree to the Reddit Brand Terms of Use and follow the guidelines in this document, you’re good to go. In most other cases, you’ll need to get our permission first.

Related PR for social-logos: https://github.com/Automattic/social-logos/pull/87

Fixes #9819

Internal reference: p7fD6U-2xE-p2

#### Changes proposed in this Pull Request:

* Updates social-logos assets to update style of reddit icon

#### Testing instructions:

* Make sure that sharing buttons are enabled for your site and reddit button is included. Select "Icon only" buttons type.
* Check any post, make sure sharing buttons are there, and reddit icon matches the acutal design.
* Change buttons type to "Icon & Text". Check the sharing button on your site.

One thing I not sure is the size of the icon. It looks fine in "Icon Only", but looks too big in "Icon & Text". Also `.sd-social-icon .sd-content ul li[class*='share-'] a.sd-button:before` class adds a `top: 1px` which put "Icons Only" icons off-center

**Before**

Hello world! – HelloWord 2019-05-23 15-44-29
before2

**After**

Hello world! – HelloWord 2019-05-23 15-31-10
Hello world! – HelloWord 2019-05-23 15-35-00

#### Proposed changelog entry for your changes:

* update reddit icon

### CODE REVIEW

**Review by jeherve — COMMENTED** [review]
This tests well for me, but I am a bit worried seeing all those selectors change.

File: _inc/social-logos/social-logos.css [hunk 1]
```diff
@@ -37,221 +39,226 @@
 /* Icons */
 
 
-.social-logo__amazon:before {
```
[the remark this answers is not in the document]
Comment (brbrr) [reply _inc/social-logos/social-logos.css]: I think that social-logos build script got updated: https://github.com/Automattic/social-logos/commit/d35743ca5e23f7bfe928f6bc6ad90e9e1d4dbe2d#diff-35b4a816e0441e6a375cd925af50752cL124

[the remark this answers is not in the document]
Comment (brbrr) [reply _inc/social-logos/social-logos.css]: It seems any of these classes not used anywhere, so I'd say we fine with this change.

**Comment by eeeeevon13** [comment]
Seems to work for me as well

**Comment by jeherve** [comment]
> These changes might need to be confirmed wih them. 

Internal reference: p7fD6U-2xE-p2

**Comment by kraftbj** [comment]
Closing per p7fD6U-2xE-p2#comment-8918

**Pull request closed without merging**